test/test_selectors.py

```diff
@@ -0,0 +1,672 @@
+            self.assertEqual(1, len(s.select(timeout=0)))
+            self.assertEqual(1, len(s.select(timeout=-1)))
+            self.assertEqual(1, len(s.select(timeout=0.001)))
+
+    def test_select_timeout_not_ready(self):
+        s = self.make_selector()
+        rd, wr = self.make_socketpair()
+        s.register(rd, selectors.EVENT_READ)
+
+        with self.assertTakesTime(upper=SHORT_SELECT):
+            self.assertEqual(0, len(s.select(timeout=0)))
+
+        with self.assertTakesTime(lower=SHORT_SELECT):
+            self.assertEqual(0, len(s.select(timeout=SHORT_SELECT)))
+
+    @skipUnless(hasattr(signal, "alarm"), "Platform doesn't have signal.alarm()")
+    def test_select_timing(self):
+        s = self.make_selector()
+        rd, wr = self.make_socketpair()
+        key = s.register(rd, selectors.EVENT_READ)
+
+        sigalrm_handler = signal.signal(signal.SIGALRM, lambda *args: wr.send(b'x'))
+        self.addCleanup(signal.signal, signal.SIGALRM, sigalrm_handler)
+
+        # Start the timer for the interrupt.
+        self.set_alarm(SHORT_SELECT)
+
+        with self.assertTakesTime(upper=LONG_SELECT):
+            ready = s.select(LONG_SELECT)
+        self.assertEqual([(key, selectors.EVENT_READ)], ready)
+
+    @skipUnless(hasattr(signal, "alarm"), "Platform doesn't have signal.alarm()")
+    def test_select_interrupt_no_event(self):
+        s = self.make_selector()
+        rd, wr = self.make_socketpair()
+        s.register(rd, selectors.EVENT_READ)
+
+        self.make_alarm(lambda *args: None, SHORT_SELECT)
+
+        # The x1.1 is because SelectSelector is really slow on CI.
+        with self.assertTakesTime(lower=LONG_SELECT, upper=LONG_SELECT * 1.1):
+            self.assertEqual([], s.select(LONG_SELECT))
+
+    @skipUnless(hasattr(signal, "alarm"), "Platform doesn't have signal.alarm()")
+    def test_select_interrupt_with_event(self):
+        s = self.make_selector()
+        rd, wr = self.make_socketpair()
+        s.register(rd, selectors.EVENT_READ)
+        key = s.get_key(rd)
+
+        self.make_alarm(lambda *args: wr.send(b'x'), SHORT_SELECT)
+
+        with self.assertTakesTime(lower=SHORT_SELECT, upper=SHORT_SELECT):
+            self.assertEqual([(key, selectors.EVENT_READ)], s.select(LONG_SELECT))
+        self.assertEqual(rd.recv(1), b'x')
+
+    @skipUnless(hasattr(signal, "alarm"), "Platform doesn't have signal.alarm()")
+    def test_select_multiple_interrupts_with_event(self):
+        s = self.make_selector()
+        rd, wr = self.make_socketpair()
+        s.register(rd, selectors.EVENT_READ)
+        key = s.get_key(rd)
+
+        def second_alarm(*args):
+            wr.send(b'x')
+
+        def first_alarm(*args):
+            self.set_alarm(SHORT_SELECT)
+            signal.signal(signal.SIGALRM, second_alarm)
+
+        self.make_alarm(first_alarm, SHORT_SELECT)
+
+        with self.assertTakesTime(lower=SHORT_SELECT * 2, upper=SHORT_SELECT * 4):
```

Comment:
Probably not, I will try with * 3 because I suspect that alarm handling on the CI is what is so slow. :/